### ISSUE TITLE
View Stackdriver logs

### ISSUE THREAD

**Issue by grant** [google/clasp#46]
## Expected Behavior

```sh
clasp logs
(DEBUG) {foo:42} – myFunction [2018-02-21T16:08:06.364Z]
```

```sh
clasp logs --json
[{
  "jsonPayload": {foo:42},
  ...
}, {...}]
```

Views StackDriver logs (`console.log`).

## Actual Behavior

Command doesn't exist yet.

## Notes

- [Stackdriver API (entries/list)](https://cloud.google.com/logging/docs/reference/v2/rest/v2/entries/list)
- [Apps Script console.log](https://developers.google.com/apps-script/reference/base/console#methods)

### Example Stackdriver API result

```json
{
  "insertId": "13oj0ajfyb6tdx",
  "jsonPayload": {foo:42},
  "labels": {…},
  "logName": "projects/project-id-4088669541713651222/logs/script.googleapis.com%2Fconsole_logs",
  "receiveTimestamp": "2018-02-21T16:08:07.370458993Z",
  "resource": {
    "labels": {
      "function_name": "myFunction",
      "invocation_type": "editor",
      "project_id": "project-id-4088669541713651222",
    },
    "type": "app_script_function"
  },
  "severity": "DEBUG",
  "timestamp": "2018-02-21T16:08:06.364Z",
}
```

Please 👍 if you'd like this feature.

**Comment by campionfellin** [comment]
I'd like to take this on!
I have written a basic logs function, which you can see here:

https://github.com/campionfellin/clasp/commit/01b0b42623ef9784874cd7daade86587ab151fd3

It needs some cleanup and styling, but should work.

**Comment by grant** [comment]
Sweet! Looking good!

1 nit. We'll probably want to use the `ProjectSettings` instead of an env variable. Like:
```ts
// Project settings file (Saved in .clasp.json)
interface ProjectSettings {
  scriptId: string;
  rootDir: string;
  projectId: string; // Resources -> Cloud Platform Project...
}
```

**Comment by campionfellin** [comment]
Yes, I saw the ProjectSettings, and I wasn't sure how we wanted to implement actually getting the projectId.

I see a couple options:

1) Make a command like `clasp config set projectId=test123`
       This would likely mean making other subcommands under `clasp config`
2) Run `gcloud projects list` for the user, and allow them to choose the project they have associated with the Google Script.

For me, (1) seems like the easier way to do it, and maybe what we should do temporarily. In the future, I think that maybe adding a flag to `clasp create` like `--enable-stackdriver-logs` would be the best option. During the create process we could then automatically create a GCP project for them with Stackdriver logs enabled and save the projectId in the ProjectSettings.

**Comment by grant** [comment]
Currently, the other project settings (`scriptId` and `rootDir`) are in the `.clasp.json` file. Let's just add there.

---

I think a `clasp config` method could be a future option since we're adding a few properties now, but initially let's have users just modify the json file.

Also, come to think of it, we should add instructions to the README.

So, for v1, assume the Cloud `projectId` is already set in project settings and we just need to use it. Ideally, the API would give us the Cloud `projectId` for the [Project](https://developers.google.com/apps-script/api/reference/rest/v1/projects#Project) (on `create` like you said).

**Comment by campionfellin** [comment]
Ok, I'll assume for now that it's already in the .clasp.json file, with instructions to the user to modify that file.

In the README instructions, we should mention that you can find your projectId by following these instructions: https://developers.google.com/apps-script/guides/cloud-platform-projects#accessing_an_apps_script_cloud_platform_project

However, it's important to note that this is a **hidden** GCP project, and is only used for the Apps Script. If they want to connect it to another project they can follow the instructions a little lower on that page as well.

+1 to "Ideally, the API would give us the Cloud projectId for the Project". Since it doesn't for now, I think updating the README with how to find it and where to put it would be best.

**Comment by campionfellin** [comment]
@grant See the latest 3 commits here: https://github.com/campionfellin/clasp/commits/clasp-logs

**Comment by grant** [comment]
Fixed with #101.
